Hd md dev 2023 09 18

JFRudzinski opened the pull request:
@JosePizarro3 I have finished reorganizing into the flatter structure that we discussed this morning. I probably still need to fix some links and such, but let me know if you agree with the basic premise.

JFRudzinski committed:
fixes from Abbas review

JFRudzinski committed:
added observables

JFRudzinski committed:
reorganized to follow diataxis, added TOCs on all pages

JosePizarro3 commented:
I didn't forget to review this, probably will do it tomorrow or on Fri :-)

JFRudzinski commented:
> I didn't forget to review this, probably will do it tomorrow or on Fri :-)

No rush really. I will let you know at any point if it is important, but otherwise feel free to prioritize other things.

JosePizarro3 commented on docs/custom_schemas/h5md/howto-H5MD-NOMAD.md:
```diff
@@ -0,0 +1,811 @@
+[h5py](https://docs.h5py.org/en/stable/quick.html)
+: module for reading and writing HDF5 files.
+
+[UnitRegistry](https://pint.readthedocs.io/en/0.10.1/tutorial.html)
```
Why not using [`from nomad.units import ureg`](https://gitlab.mpcdf.mpg.de/nomad-lab/nomad-FAIR/-/tree/develop/nomad/units?ref_type=heads)? Or at this point you don't want them to have NOMAD installed?

JFRudzinski replied:
yeah, exactly, but I now also added this alternative for people who have already installed nomad-lab

JosePizarro3 commented on docs/custom_schemas/h5md/howto-H5MD-NOMAD.md:
```diff
@@ -0,0 +1,811 @@
+
+### Example Data
+
+For concreteness, we consider a fictitious set of "vanilla" molecular dynamics simulations, run with the OpenMM software. The following definitions set the dimensionality, periodicity, and the units for this simulation.
```
Add href to OpenMM: https://openmm.org/

JFRudzinski replied:
done.

JosePizarro3 commented on docs/custom_schemas/h5md/howto-H5MD-NOMAD.md:
````diff
@@ -0,0 +1,811 @@
+acceleration_unit = 1.0 * length_unit / time_unit**2
+```
+
+In this example, we will assume that the relevant simulation data is compatible with [MDAnalysis](https://www.mdanalysis.org/), such that a `universe` containing the trajectory and topology information can be created.
````
I'd use **universe** (bolding) instead of `universe` (coding)

JFRudzinski replied:
done.

JosePizarro3 commented on docs/custom_schemas/h5md/howto-H5MD-NOMAD.md:
````diff
@@ -0,0 +1,811 @@
+```
+Here `n_bonds`, `n_angles`, `n_dihedrals`, and `n_impropers` represent the corresponding number of instances of each interaction within the force field.
+
+The creation of the `particles_group` group (i.e., topology) is discussed [HERE](#creating-a-topology-particles_group)
````
"You can read more about the creation of the hierarchical `particles_group` in [Creating the standard hierarchy from a MDAnalysis universe](#creating-a-topology-particles_group)."

JFRudzinski replied:
done.

JosePizarro3 commented on docs/custom_schemas/h5md/howto-H5MD-NOMAD.md:
```diff
@@ -0,0 +1,811 @@
+
+### [Parameter Group](explanation-H5MD-NOMAD.md#the-parameters-group)
+
+Using the json templates for [force calculations](../parameters.md#force_calculation_template_anchor) and [molecular dynamics workflows](explanation-H5MD-NOMAD.md#the-observables-group#md_workflow_template_anchor), the (meta)data can be written to the H5MD-NOMAD file using the following code:
```
The first link to force calculations is broken. I think you just have to delete the "../" from the path.

Furthermore, the molecular dynamics workflows link takes you to the general explanation, but not to the specific section you might wanna be referencing to.

JFRudzinski replied:
done.

JosePizarro3 commented on docs/custom_schemas/h5md/howto-H5MD-NOMAD.md:
```diff
@@ -0,0 +1,811 @@
+
+Using the json templates for [force calculations](../parameters.md#force_calculation_template_anchor) and [molecular dynamics workflows](explanation-H5MD-NOMAD.md#the-observables-group#md_workflow_template_anchor), the (meta)data can be written to the H5MD-NOMAD file using the following code:
+
+First, import the templates:
```
"First, import the data extracted from the JSON templates:"

JFRudzinski replied:
done.

JosePizarro3 commented on docs/custom_schemas/h5md/howto-H5MD-NOMAD.md:
````diff
@@ -0,0 +1,811 @@
+workflow = get_parameters_recursive(workflow, workflow_parameters)
+```
+
+It's as simple as that! Now, we can [upload our H5MD-NOMAD file directly to NOMAD](../../uploading_and_publishing_data/overview.md) and all the written (meta)data will be stored according to the standard NOMAD schema.
````
href broken too. Probably it should be: "../uploading_and_publishing/overview.md"

JFRudzinski replied:
fixed.

JosePizarro3 commented on docs/custom_schemas/h5md/howto-H5MD-NOMAD.md:
```diff
@@ -0,0 +1,811 @@
+
+## Creating a topology (`particles_group`)
+
+This page demonstrates how to create a "standard" topology in H5MD-NOMAD. The demonstrated organization of molecules and monomers is identical to what other NOMAD parsers do to create a topology from native simulation files (e.g., outputs from Gromacs or Lammps). However, the user is free to deviate from this standard to create arbitrary organizations of particles, as described in [Connectivity]( ...
```
Capital letters for codes (GROMACS and LAMMPS), and it will be nice some references: one for the "writing a parser plugin" section when you say "NOMAD parsers", and some hrefs for the code pages themselves.

JFRudzinski replied:
changed to capitals, added refs for code websites. Added TODO for linking to other sections in the NOMAD docs later

JosePizarro3 commented on docs/custom_schemas/h5md/howto-H5MD-NOMAD.md:
````diff
@@ -0,0 +1,811 @@
+└── ...
+```
+Here, the first level of organization is the "molecule group". Molecule groups contain molecules of the same type. In other words, `molecule_group_1` and `molecule_group_2` represent distinct molecule types. At the next level of the hierarchy, each molecule within this group is stored (i.e., `molecule_1`, `molecule_2`, etc.). In the above example, `molecule_group_1` represents a polymer (or prote ...
+<!-- TODO - add image of topology bar in overview page? Here or maybe when the topology is introduced on the connectivity page? -->
````
Adding to your TODO, I think there should be a full explanation and reference about what Topology/Groups mean in NOMAD. Just so we know that we have to add it.

JFRudzinski replied:
done.

JosePizarro3 commented on docs/custom_schemas/h5md/explanation-H5MD-NOMAD.md:
```diff
@@ -0,0 +1,1342 @@
+
+H5MD was originally proposed by P. de Buyl, P. H. Colberg and F. Höfling in [H5MD: A structured, efficient, and portable file format for molecular data](http://dx.doi.org/10.1016/j.cpc.2014.01.018), Comp. Phys. Comm. 185, 1546–1553 (2014) [[arXiv:1308.6382](http://arxiv.org/abs/1308.6382)]. The schema is maintained, along with associated tools, in a GitHub repository: [H5MD GitHub](https://github ...
+
+This section provides the basic nomenclature of the H5MD schema relevant for understanding H5MD-NOMAD, and was duplicated or summarized from the [H5MD webpage](http://h5md.nongnu.org/).
```
...were duplicated...

Right? :smile:

JFRudzinski replied:
I think no actually: "This section...was duplicated..."

JosePizarro3 commented on docs/custom_schemas/h5md/explanation-H5MD-NOMAD.md:
```diff
@@ -0,0 +1,1342 @@
+
+## Overview
+
+Most computational data in NOMAD is harvested with code-specific parsers that recognize the output files from a particular software and retrieve the appropriate (meta)data accordingly.
```
I cannot comment too much on this explanation, but I thought that the role of NOMAD was mostly giving data storage support to what you store using H5MD. From this perspective, I would say that there is this H5MD standard, and we are also developing the mapping to such standard into the NOMAD database. With the ultimate goal to give the other way around, and be able to support H5MD standarization from the NOMAD archive data.

JFRudzinski replied:
Not sure if I understand your comment, but I changed the last sentence in this paragraph to say that H5MD-NOMAD allows storage of any simulation data in NOMAD.

JosePizarro3 replied:
The second phrase, should have been "are also developing the mapping to such standard into the NOMAD **archive**".

I think I meant that what we want is to map a H5MD file into the NOMAD archive, to then eventually (maybe, also something similar with NeXuS?) go the other way around with a _mapper_.

But yes, the change looks good.

JosePizarro3 commented on docs/custom_schemas/h5md/explanation-H5MD-NOMAD.md:
```diff
@@ -0,0 +1,1342 @@
+    type.
+
+`image`
+:   <a id="image_anchor"></a>**(currently unused in H5MD-NOMAD)**
```
I think you can delete this image_anchor element, right?

JFRudzinski replied:
Actually it's linked in the reference.

JosePizarro3 commented on docs/custom_schemas/h5md/explanation-H5MD-NOMAD.md:
```diff
@@ -0,0 +1,1342 @@
+`image`
+:   <a id="image_anchor"></a>**(currently unused in H5MD-NOMAD)**
+
+??? details
```
I like the idea of these admonitions, but the issue is that visually they are not so obviously linked with the quantity you are defined. I will keep it simply and add this description after the bold parenthesis, or directly, not adding any explanation.

JFRudzinski replied:
I made a TODO about this. I agree with your comment, but I really want this information to be hidden. I guess an alternative would be to link to the corresponding section of the original H5MD docs, but I'm not sure how persistent that is.

JosePizarro3 commented on docs/custom_schemas/h5md/explanation-H5MD-NOMAD.md:
```diff
@@ -0,0 +1,1342 @@
+:   additional metadata may be given as necessary.
+<!-- TODO - Is this really parsed?! -->
+
+A list of standardized observables can be found [HERE](reference-H5MD-NOMAD.md#standardized-observables-in-h5md-nomad).
```
Maybe instead of "HERE", use directly the name of the section, something like "Reference - H5MD-NOMAD > Standardized observables in H5MD-NOMAD"

JFRudzinski replied:
done.

JosePizarro3 commented on docs/custom_schemas/h5md/explanation-H5MD-NOMAD.md:
```diff
@@ -0,0 +1,1342 @@
+     |    \-- ...
+     \-- ...
+
+The subgroups `force_calculations` and `workflow` are supported. The following describes the detailed data structures for these subgroups, using the NOMAD MetaInfo definitions for each underlying `Quantity`. Please note that:
```
Maybe adding href to Glossary? If Quantity does not exist in Glossary, feel free to create it.

JFRudzinski replied:
added a TODO for now. I guess I would like to wait until I add this to the main docs and then develop these links there.

JosePizarro3 commented on docs/custom_schemas/h5md/explanation-H5MD-NOMAD.md:
```diff
@@ -0,0 +1,1342 @@
+
+1. If needed, the leading prefactor is stored as a separate attribute of `float` datatype called `unit_factor`.
+
+2. The string that describes the unit should be compatible with the `UnitRegistry` class of the `pint` python module.
```
Maybe explain that NOMAD has its own definitions in nomad.units following the UnitRegistry from pint? I am not 100% sure if pint covers all the possibilities as we do (i.e., I am not sure if nomad/units/default_en.txt and everything defined there is covered by pint, or there are conversions that nomad does which are used for it).

JFRudzinski replied:
I am also not sure, but I think we are safe in the sense that I only am asking that the strings are compatible with the pint package. Also, there is no documentation really about the list of NOMAD unit strings, but there is on the pint docs. And I would sort of like to stay as agnostic to NOMAD as possible, so people aren't overwhelmed by trying to look into NOMAD docs.

JosePizarro3 reviewed (commented):
Pretty good. All is minor stuff.

Furthermore, I was thinking whether it will be better to split the current How-to into 3 How-tos with the structure you are following, i.e., "How-to write an HDF5 file compliant with the H5MD-NOMAD standard", "How-to access the H5MD-NOMAD file", "How to create a NOMAD topology". You decide.

JosePizarro3 reviewed (commented):
Pretty good. All is minor stuff.

Furthermore, I was thinking whether it will be better to split the current How-to into 3 How-tos with the structure you are following, i.e., "How-to write an HDF5 file compliant with the H5MD-NOMAD standard", "How-to access the H5MD-NOMAD file", "How to create a NOMAD topology". You decide.

JFRudzinski committed:
changes from Chema's review

JFRudzinski committed:
a few last changes

JFRudzinski commented:
@JosePizarro3  I have addressed all your comments now. For a few of them I just made TODO's to deal with later. I think I would like to merge this before you leave. It is not urgent if you disagree, but since we're going to work some on the general docs next week I thought it would be nice for everything we have to be merged for reference.

JosePizarro3 commented:
Sure, go ahead. Merge at will.
